contracts/seed/Seed.sol

```diff
@@ -163,37 +165,34 @@ contract Seed {
         minimumReached  = false;
         maximumReached  = false;
         seedRemainder   = IERC20(_tokens[0]).balanceOf(address(this));
+        allocatedSeed   = seedRemainder;
     }
 
     /**
-      * @dev                     Buy seed tokens.
+      * @dev                     Buy and lock seed tokens.
       * @param _seedAmount       The amount of seed tokens to buy.
     */
     function buy(uint256 _seedAmount) public isActive allowedToBuy {
-        seedRemainder = seedRemainder.sub(_seedAmount);
         //  fundingAmount is an amount of fundingTokens required to buy _seedAmount of SeedTokens
         uint256 fundingAmount = (_seedAmount.mul(price)).div(PCT_BASE);
+
         // Funding Token balance of this contract;
         uint256 fundingBalance = fundingToken.balanceOf(address(this));
-        //  alreadyLockedSeedTokens is an amount of already locked SeedTokens without fee
-        uint256 alreadyLockedSeedTokens = (fundingBalance.mul(PCT_BASE)).div(price);
+
         //  feeAmount is an amount of fee we are going to get in seedTokens
         uint256 feeAmount = (_seedAmount.mul(uint256(PPM))).mul(fee).div(PPM100);
-        //  lockedSeedFee is an amount of fee we already need to pay in seedTokens
-        uint256 lockedSeedFee = (alreadyLockedSeedTokens.mul(uint256(PPM))).mul(fee).div(PPM100);
 
         // total fundingAmount should not be greater than the hardCap
         require( fundingBalance.
                   add(fundingAmount).
                   add((feeAmount.mul(price)).div(PCT_BASE)) <= hardCap,
             "Seed: amount exceeds contract sale hardCap");
 
-        // We are calculating that we are not exceeding balance of seedTokens in this contract
-        // balanceToBe = amountOfSeedAlreadyLocked + amountOfSeedToLock + SeedFee
-        require( seedToken.balanceOf(address(this)) >= (alreadyLockedSeedTokens.
-                                                        add(_seedAmount).
-                                                        add(feeAmount)),
-            "Seed: seed distribution exceeded");
+        // the amount of seed tokens still to be distributed
+        seedRemainder = (seedRemainder.sub(_seedAmount)).sub(feeAmount);
+
+        // require( seedToken.balanceOf(address(this)) >= seedRemainder,
```

Comment:
pls remove the comment

Reply:
I would say, pls, add the correct requirement statement for the seedToken balance check

Reply:
> I would say, pls, add the correct requirement statement for the seedToken balance check

+1

Reply:
I don't think we need check for seed token balance. If we see, we don't have any other way to transfer seedToken without claimLock().

Reply:
Have added the require statement.